modules/internal/ChapelDomain.chpl

```diff
@@ -1615,15 +1610,28 @@ module ChapelDomain {
         return idx;
     }
 
+    pragma "last resort"
+    proc orderToIndex(order) {
+      if this.isRectangular() && isNumericType(this.idxType) then
+        compilerError("the order argument of 'orderToIndex' on this domain must be an integer, excluding uint(64)");
```

Comment:
Wording: "illegal value passed to orderToIndex(): the argument 'order' must be an integer not exceeding the size of 'int(64)'"

Reply:
To me "not exceeding the size of 'int(64)'" sounds vague. I am implementing the rest of these suggestions.

Reply:
Sounds good